pageserver/src/metrics.rs

```diff
@@ -1394,15 +1395,48 @@ impl<F: Future<Output = Result<O, E>>, O, E> Future for MeasuredRemoteOp<F> {
 }
 
 pub fn preinitialize_metrics() {
-    // We want to alert on this metric increasing.
-    // Initialize it eagerly, so that our alert rule can distinguish absence of the metric from metric value 0.
-    assert_eq!(UNEXPECTED_ONDEMAND_DOWNLOADS.get(), 0);
-    UNEXPECTED_ONDEMAND_DOWNLOADS.reset();
+    // Python tests need these and on some we do alerting.
+    //
+    // FIXME: make it so that we have no top level metrics as this fn will easily fall out of order:
+    // - global metrics reside in a Lazy<PageserverMetrics>
+    //   - access via crate::metrics::PS_METRICS.materialized_page_cache_hit.inc()
+    // - could move the statics into TimelineMetrics::new()?
+
+    // counters
+    [
+        &MATERIALIZED_PAGE_CACHE_HIT,
+        &MATERIALIZED_PAGE_CACHE_HIT_DIRECT,
+        &UNEXPECTED_ONDEMAND_DOWNLOADS,
+        &WALRECEIVER_STARTED_CONNECTIONS,
+        &WALRECEIVER_BROKER_UPDATES,
+        &WALRECEIVER_CANDIDATES_ADDED,
+        &WALRECEIVER_CANDIDATES_REMOVED,
+    ]
+    .into_iter()
```

Comment:
This would be easier with the function instead of a closure: `.for_each(Lazy::get)`. Or alternatively, a for loop like `let counters = [ ... ]; for c in counters`.

Reply:
Was this the one with the started timer which gets discarded? If so, can change it.

Reply:
`.for_each(Lazy::get)` does not work because it returns non-unit

How is a for loop easier? Surely you'd have to do something within the for-loop body?

Reply:
Yeah but I think you save a line at least. Shrug. Fair points about `Lazy::get` not working. Consider my concerns resolved.

Reply:
Well, since @skyzh also wants that so maybe I'll use it, rustfmt will want extra newlines for a new code block :) 

> `let counters = [ ... ]; for c in counters`

yeah this would save a line of into_iter... But it comes at a cost of having a binding :)

Reply:
[f18647a](https://github.com/neondatabase/neon/pull/4801/commits/f18647ab99aacc0e0bebc9f95181c5517a449d53)